WalletWasabi.Fluent/ViewModels/Wallets/Send/PrivacySuggestionsFlyoutViewModel.cs

```diff
@@ -36,45 +44,68 @@ public PrivacySuggestionsFlyoutViewModel()
 
 	public ObservableCollection<SuggestionViewModel> Suggestions { get; }
 
+	/// <remarks>Method supports being called multiple times. In that case the last call cancels the previous one.</remarks>
 	public async Task BuildPrivacySuggestionsAsync(Wallet wallet, TransactionInfo info, BitcoinAddress destination, BuildTransactionResult transaction, bool isFixedAmount, CancellationToken cancellationToken)
 	{
-		_suggestionCancellationTokenSource?.Cancel();
-		_suggestionCancellationTokenSource?.Dispose();
+		using CancellationTokenSource singleRunCts = new();
 
-		_suggestionCancellationTokenSource = new(TimeSpan.FromSeconds(15));
-		using CancellationTokenSource linkedCts = CancellationTokenSource.CreateLinkedTokenSource(_suggestionCancellationTokenSource.Token, cancellationToken);
+		lock (_lock)
+		{
+			_suggestionCancellationTokenSource?.Cancel();
+			_suggestionCancellationTokenSource = singleRunCts;
+		}
 
-		Suggestions.Clear();
-		SelectedSuggestion = null;
+		using CancellationTokenSource timeoutCts = new(TimeSpan.FromSeconds(15));
+		using CancellationTokenSource linkedCts = CancellationTokenSource.CreateLinkedTokenSource(timeoutCts.Token, singleRunCts.Token, cancellationToken);
```

Comment:
```suggestion
		using CancellationTokenSource linkedCts = CancellationTokenSource.CreateLinkedTokenSource(timeoutCts.Token, _suggestionCancellationTokenSource.Token, cancellationToken);
```

Since they point to the same cts, because of LOC55.

Probably doesn't make any difference, so feel free to ignore.